package.json

```diff
@@ -38,7 +38,7 @@
     "exports": {
       ".": {
         "import": {
-          "types": "./dist/index.d.mts",
+          "types": "./dist/index.d.ts",
```

Comment:
Please note that TypeScript type declarations have "types"; there's the CommonJS type declaration (`.d.cts`) and the ESM type declaration (`.d.mts`). 

We are checking them with https://arethetypeswrong.github.io.

Reply:
Are there specific tools that actually require the types for the ESM and CJS builds to be separate?

When I looked at the emmitted types with the tsup setup, I found that there were:

- `.d.ts` for CommonJS
- `.d.mts` for ESM

And checking a bunch of them, they were always identical. So having both seemed like needless duplication.

Reply:
Answering myself:

> It’s important to note that the CommonJS entrypoint and the ES module entrypoint each needs its own declaration file, even if the contents are the same between them. Every declaration file is interpreted either as a CommonJS module or as an ES module, based on its file extension and the "type" field of the package.json, and this detected module kind must match the module kind that Node will detect for the corresponding JavaScript file for type checking to be correct. Attempting to use a single .d.ts file to type both an ES module entrypoint and a CommonJS entrypoint will cause TypeScript to think only one of those entrypoints exists, causing compiler errors for users of the package.

From https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing

Reply:
You're right. Although the contents might be identical, it can bring a potential bug to our codebase; it is better for us to split the type declarations.